src/praisonai-agents/praisonaiagents/memory/memory.py

```
@@ -980,6 +1003,27 @@ def search_long_term(
+                    adapter_results = [
+                        r for r in adapter_results
+                        if r.get("metadata", {}).get("quality", 0.0) >= min_quality
+                    ]
+                if relevance_cutoff > 0:
+                    adapter_results = [r for r in adapter_results if r.get("score", 1.0) >= relevance_cutoff]
+                final_results = adapter_results[:limit]
+                top_score = final_results[0].get("score") if final_results else None
+                self._emit_memory_event("search", "long_term", query=query,
+                                       result_count=len(final_results), top_score=top_score)
+                return final_results
+            except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)